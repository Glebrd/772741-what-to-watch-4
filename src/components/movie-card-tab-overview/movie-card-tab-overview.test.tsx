import * as React from "react";
import * as renderer from "react-test-renderer";
import MovieCardTabOverview from "./movie-card-tab-overview";

const movie = {
  id: 634,
  title: `The Truman Show`,
  picture: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  genre: `Comedy`,
  date: `1998`,
  poster: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  background: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  rating: 7.2,
  scores: 941,
  director: `Louise Virgoe`,
  starring: [`Pamela Anthiftle`, `Arlene Cordelle`, `Arleen Portriss`, `Christy Hullin`],
  description: `An insurance salesman discovers his whole life is actually a reality TV show.`,
  videoPreview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  runTime: `1h 43m`,
};

const {rating, description, starring, director, scores} = movie;

it(`Movie card tab overview component renders correctly`, () => {
  const tree = renderer
    .create(
        <MovieCardTabOverview
          rating={rating}
          description={description}
          starring={starring}
          director={director}
          scores={scores}
        />).toJSON();

  expect(tree).toMatchSnapshot();
});
