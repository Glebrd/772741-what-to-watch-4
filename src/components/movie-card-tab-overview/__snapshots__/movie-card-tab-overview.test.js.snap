// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie card tab overview component renders 1`] = `
Array [
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      7.2
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Good
      </span>
      <span
        className="movie-rating__count"
      >
        941
         ratings
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text"
  >
    <p>
      An insurance salesman discovers his whole life is actually a reality TV show.
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        Louise Virgoe
         
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        Pamela Anthiftle, Arlene Cordelle, Arleen Portriss, Christy Hullin
         
      </strong>
    </p>
  </div>,
]
`;
